tiledb/sm/array_schema/array_schema.h

```diff
@@ -356,6 +380,23 @@ class ArraySchema {
   /** The format version of this array schema. */
   uint32_t version_;
 
+  /** Mutex for thread-safety. */
+  mutable std::mutex mtx_;
+
+  /**
+   * The timestamp the array schema was written.
+   * This is used to determine the array schema file name.
+   * The two timestamps are identical.
+   * It is stored as a pair to keep the usage consistent with metadata
+   */
+  std::pair<uint64_t, uint64_t> timestamp_range_;
+
+  /** The URI of the array schema file. */
+  URI uri_;
+
+  /** The file name of array schema in the format of timestamp_timestamp_uuid */
```

Comment:
Noting here that for the fragment metadata folders, the format is `t_t_uuid_v`, where `v` is the file format version. Shall we follow the same convention here?

Reply:
Not sure about if we sould like to add schema version to make the format consistent with fragment metadata folders. I will discuss that with others.

Reply:
What if the array schema file format changes though? Will we be able to know how to read the file?

Reply:
Actually I take this back. For the array schema we read the whole file from storage, then retrieve the version first and then we can parse the rest of the file accordingly.

Reply:
I will leave the file format as-is. If we need to add version information to the filename in future, I can add them then